src/libfetchers/tarball.cc

```diff
@@ -220,8 +225,12 @@ struct CurlInputScheme : InputScheme
             if (auto n = string2Int<uint64_t>(*i))
                 input.attrs.insert_or_assign("revCount", *n);
 
-        url.query.erase("rev");
-        url.query.erase("revCount");
+        if (auto i = get(url.query, "lastModified"))
+            if (auto n = string2Int<uint64_t>(*i))
+                input.attrs.insert_or_assign("lastModified", *n);
+
+        for (auto & param : allowedAttrs())
+            url.query.erase(param);
```

Comment:
This is perhaps good, but unrelated?

Reply:
Yeah it's unrelated. It makes it more similar to the Git/GitHub input schemes, which also have an `allowedAttrs()`.

Reply:
I don't see `allowedAttrs` used in `inputFromURL` in `github.cc` however?

Reply:
Right, but the GitHub input type doesn't have a URL from which we have to remove the Nix-specific query parameters.

Reply:
Oh, I finally understand what is going on!